fe/fe-core/src/main/java/com/starrocks/backup/Repository.java

```diff
@@ -702,22 +710,33 @@ public void readFields(DataInput in) throws IOException {
         createTime = in.readLong();
 
         if (!GlobalStateMgr.isCheckpointThread()) {
-            // check __palo_repository_ first, if success, prefixRepo = __palo_repository_
-            String listPath = Joiner.on(PATH_DELIMITER).join(location, joinPrefix("__palo_repository_", name));
-            Status st;
-            try {
-                st = storage.checkPathExist(listPath);
-            } catch (Exception e) {
-                LOG.warn("check path exist fail");
-                prefixRepo = "__starrocks_repository_";
-                return;
-            }
+            genPrefixRepo();
+        }
+    }
 
-            if (st.ok()) {
-                prefixRepo = "__palo_repository_";
-            } else {
-                prefixRepo = "__starrocks_repository_";
-            }
+    @Override
+    public void gsonPostProcess() throws IOException {
+        if (!GlobalStateMgr.isCheckpointThread()) {
+            genPrefixRepo();
+        }
+    }
+
+    private void genPrefixRepo() {
+        // check __palo_repository_ first, if success, prefixRepo = __palo_repository_
+        String listPath = Joiner.on(PATH_DELIMITER).join(location, joinPrefix("__palo_repository_", name));
+        Status st;
+        try {
+            st = storage.checkPathExist(listPath);
+        } catch (Exception e) {
+            LOG.warn("check path exist fail");
+            prefixRepo = "__starrocks_repository_";
+            return;
+        }
+
+        if (st.ok()) {
+            prefixRepo = "__palo_repository_";
```

Comment:
can we remove this completely and if not, why?

Reply:
We should keep this logic, the change from __palo_repository_ to __starrocks_repository_ is started from version 3.0